### ISSUE TITLE
I have problems running this extension

### ISSUE THREAD

**Issue by tgagor** [sensu-extensions/sensu-extensions-influxdb#7]
Hi,

I was able to configure this extension and looks like events are properly forwarded to handler influxdb but it's crashing for every event like that:

```
{
  "status": 2,
  "output": "NoMethodError: undefined method `gsub' for nil:NilClass\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-extensions-influxdb-0.0.2/lib/sensu/extensions/influxdb.rb:141:in `sanitize'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-extensions-influxdb-0.0.2/lib/sensu/extensions/influxdb.rb:157:in `parse_line'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-extensions-influxdb-0.0.2/lib/sensu/extensions/influxdb.rb:59:in `block in run'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-extensions-influxdb-0.0.2/lib/sensu/extensions/influxdb.rb:56:in `each'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-extensions-influxdb-0.0.2/lib/sensu/extensions/influxdb.rb:56:in `run'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-extension-1.5.0/lib/sensu/extension.rb:113:in `safe_run'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-0.26.5/lib/sensu/server/handle.rb:120:in `handler_extension'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-0.26.5/lib/sensu/server/handle.rb:146:in `handler_type_router'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-0.26.5/lib/sensu/server/handle.rb:163:in `handle_event'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-0.26.5/lib/sensu/server/process.rb:255:in `block (3 levels) in process_event'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-0.26.5/lib/sensu/server/mutate.rb:19:in `block in mutator_callback'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-extension-1.5.0/lib/sensu/extension.rb:114:in `block in safe_run'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-extensions-json-1.0.0/lib/sensu/extensions/json.rb:16:in `run'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-extension-1.5.0/lib/sensu/extension.rb:113:in `safe_run'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-0.26.5/lib/sensu/server/mutate.rb:59:in `mutator_extension'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-0.26.5/lib/sensu/server/mutate.rb:82:in `mutate_event'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-0.26.5/lib/sensu/server/process.rb:254:in `block (2 levels) in process_event'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-0.26.5/lib/sensu/server/filter.rb:303:in `block in filter_event'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-0.26.5/lib/sensu/server/filter.rb:270:in `event_filtered?'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-0.26.5/lib/sensu/server/filter.rb:301:in `filter_event'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-0.26.5/lib/sensu/server/process.rb:253:in `block in process_event'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-0.26.5/lib/sensu/server/process.rb:251:in `each'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-0.26.5/lib/sensu/server/process.rb:251:in `process_event'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-0.26.5/lib/sensu/server/process.rb:663:in `block (4 levels) in process_check_result'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-0.26.5/lib/sensu/server/process.rb:500:in `update_event_registry'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-0.26.5/lib/sensu/server/process.rb:662:in `block (3 levels) in process_check_result'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-0.26.5/lib/sensu/server/process.rb:560:in `block (3 levels) in create_event'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-0.26.5/lib/sensu/server/process.rb:468:in `event_silenced?'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-0.26.5/lib/sensu/server/process.rb:559:in `block (2 levels) in create_event'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-redis-1.6.0/lib/sensu/redis/client.rb:315:in `dispatch_response'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-redis-1.6.0/lib/sensu/redis/client.rb:335:in `parse_response_line'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-redis-1.6.0/lib/sensu/redis/client.rb:365:in `receive_data'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/eventmachine-1.2.0.1/lib/eventmachine.rb:194:in `run_machine'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/eventmachine-1.2.0.1/lib/eventmachine.rb:194:in `run'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-0.26.5/lib/sensu/server/process.rb:31:in `run'\n  /opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-0.26.5/exe/sensu-server:10:in `<top (required)>'\n  /opt/sensu/bin/sensu-server:22:in `load'\n  /opt/sensu/bin/sensu-server:22:in `<main>'",
  "extension": {
    "name": "influxdb",
    "type": "extension"
  },
  "message": "handler extension output",
  "level": "info",
  "timestamp": "2017-07-21T07:53:52.377046+0100"
}
```

It looks like after this line of code:
```
153         # Strip metric name
154         key = strip_key(key, event[:strip_metric], event[:client])
```

`key` variable is nil and it can't be further processed.

I have quite a lot of metrics on my server and switching to extension will take a lot of load from it. Please let me know how could I help.

**Comment by luisdavim** [comment]
Hi @tgagor can you please post your check configuration here? are you using the templates feature?

**Comment by Seji64** [comment]
Hi (again), i got the same error as tgagor.

```
{"timestamp":"2017-07-27T13:41:12.943083+0200","level":"info","message":"handler extension output","extension":{"type":"extension","name":"influxdb"},"event":{"id":"ef99803c-ee3f-4e93-a6ff-e554f4de17b6"},"output":"NoMethodError: undefined method `gsub' for nil:NilClass\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-extensions-influxdb-0.0.2/lib/sensu/extensions/influxdb.rb:141:in `sanitize'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-extensions-influxdb-0.0.2/lib/sensu/extensions/influxdb.rb:157:in `parse_line'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-extensions-influxdb-0.0.2/lib/sensu/extensions/influxdb.rb:59:in `block in run'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-extensions-influxdb-0.0.2/lib/sensu/extensions/influxdb.rb:56:in `each'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-extensions-influxdb-0.0.2/lib/sensu/extensions/influxdb.rb:56:in `run'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-extension-1.5.1/lib/sensu/extension.rb:113:in `safe_run'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-1.0.1/lib/sensu/server/handle.rb:128:in `handler_extension'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-1.0.1/lib/sensu/server/handle.rb:156:in `handler_type_router'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-1.0.1/lib/sensu/server/handle.rb:175:in `handle_event'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-1.0.1/lib/sensu/server/process.rb:266:in `block (3 levels) in process_event'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-1.0.1/lib/sensu/server/mutate.rb:19:in `block in mutator_callback'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-extension-1.5.1/lib/sensu/extension.rb:114:in `block in safe_run'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-extensions-json-1.0.0/lib/sensu/extensions/json.rb:16:in `run'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-extension-1.5.1/lib/sensu/extension.rb:113:in `safe_run'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-1.0.1/lib/sensu/server/mutate.rb:59:in `mutator_extension'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-1.0.1/lib/sensu/server/mutate.rb:82:in `mutate_event'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-1.0.1/lib/sensu/server/process.rb:265:in `block (2 levels) in process_event'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-1.0.1/lib/sensu/server/filter.rb:212:in `block in filter_event'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-1.0.1/lib/sensu/server/filter.rb:179:in `event_filtered?'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-1.0.1/lib/sensu/server/filter.rb:210:in `filter_event'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-1.0.1/lib/sensu/server/process.rb:264:in `block in process_event'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-1.0.1/lib/sensu/server/process.rb:262:in `each'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-1.0.1/lib/sensu/server/process.rb:262:in `process_event'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-1.0.1/lib/sensu/server/process.rb:703:in `block (4 levels) in process_check_result'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-1.0.1/lib/sensu/server/process.rb:526:in `update_event_registry'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-1.0.1/lib/sensu/server/process.rb:702:in `block (3 levels) in process_check_result'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-1.0.1/lib/sensu/server/process.rb:586:in `block (3 levels) in create_event'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-1.0.1/lib/sensu/server/process.rb:494:in `event_silenced?'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-1.0.1/lib/sensu/server/process.rb:585:in `block (2 levels) in create_event'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-redis-2.2.0/lib/sensu/redis/client.rb:337:in `dispatch_response'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-redis-2.2.0/lib/sensu/redis/client.rb:357:in `parse_response_line'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-redis-2.2.0/lib/sensu/redis/client.rb:387:in `receive_data'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/eventmachine-1.2.2/lib/eventmachine.rb:194:in `run_machine'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/eventmachine-1.2.2/lib/eventmachine.rb:194:in `run'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-1.0.1/lib/sensu/server/process.rb:33:in `run'\n  /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/sensu-1.0.1/exe/sensu-server:10:in `<top (required)>'\n  /opt/sensu/bin/sensu-server:22:in `load'\n  /opt/sensu/bin/sensu-server:22:in `<main>'","status":2}
```
My influxdb.json has the following content:
```
{
    "influxdb": {
        "host": "SIT-MONP02.mgmt.it.kle.tt",
        "port": "8086",
        "use_ssl": false,
        "database": "sensu",
        "username": "sensu",
        "password": "[PASSWORD]",
        "time_precision": "s"
    }
}
```

**Comment by Seji64** [comment]
Do you need more information? Could i provide anything else to fix this Problem?

**Comment by luisdavim** [comment]
Hi, I have a PR open that I think it covers this, can you do some tests for me?
try to set:
```
"strip_metric": "host",
```
if that solves the problem let me know and you can also try https://github.com/PTC-Global/sensu-extensions-influxdb, I think this issue is fixed there and if it is I'll merge it here.

Thanks.

**Comment by Seji64** [comment]
Great - it is working.

just a questions how to use your branch:

- Git Checkout of your branch
- Uninstall "old" Gem
- Compile your your branch
- Install your compiled gem

Is this correct?

**Comment by luisdavim** [comment]
yes, that would do it, using my fork you should be able to remove `"strip_metric": "host",`, if that's the case, I'll merge it here.
Thanks for your help.

**Comment by Seji64** [comment]
your branch is working. but i will set the `"strip_metric": "host"` tag again. cause i dont want the hostname as measurement name.

**Comment by luisdavim** [comment]
Cool. Thanks for your help.

**Comment by luisdavim** [comment]
Closed via #5
